sklearn/feature_selection/_rfe.py

```diff
@@ -512,8 +515,9 @@ class RFECV(RFE):
         :class:`~sklearn.pipeline.Pipeline` with its last step named `clf`.
 
         If `callable`, overrides the default feature importance getter.
-        The callable is passed with the fitted estimator and it should
-        return importance for each feature.
+        The callable is passed with the fitted estimator and optionally the
+        training input samples `X` and the target labels `y`. The callable
```

Comment:
Hey @thomasjpfan, thanks for reviewing my PR!

I agree that accepting two signatures makes the API more complex. On the other hand, deprecating the old signature would lead to a breaking change. In my opinion, adding a little more complexity is better than introducing a breaking change, what is your view on it?

Reply:
[Deprecating](https://scikit-learn.org/dev/developers/contributing.html#deprecation) means that the 1 parameter callable still works for the next two release. This gives users two releases to update their code to adapt to the 3 parameter callable.

Concretely, lets say we release the 3 parameter callable in v1.2, then a warning is shown stating that in v1.4, we will pass in `X` and `y` into the callable.

I prefer to deprecate the 1 parameter callable.

Reply:
Alright, now I understand what you meant by that, in that case I think that it makes sense. I will add the deprecation warning and refactor the tests to take care of that. Stay tuned!